rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/NodeSourceConfigurationParser.java

```diff
@@ -27,60 +27,73 @@
+        try {
+            return JSONParser.parseStrict(jsonStr);
+        } catch (Throwable t) {
+            LogModel logger = LogModel.getInstance();
+            logger.logCriticalMessage("JSON Parser failed with " + t.getClass().getName() + ": " + t.getMessage());
+            logger.logCriticalMessage("Raw input was: " + jsonStr);
+            return new JSONObject();
+        }
     }
 
-    public NodeSourceConfiguration parseNodeSourceConfiguration(String json) throws ImportException {
+    public NodeSourceConfiguration parseNodeSourceConfiguration(String json) {
 
         JSONObject jsonObject;
 
         try {
-            jsonObject = this.controller.parseJSON(json).isObject();
+            jsonObject = parseJSON(json).isObject();
         } catch (JSONException e) {
-            throw new ImportException("The imported node source is not a valid JSON file", e);
+            throw new IllegalArgumentException("The imported node source is not a valid JSON file", e);
         }
 
         try {
             String nodeSourceName = jsonObject.get("nodeSourceName").isString().stringValue();
             boolean nodesRecoverable = jsonObject.get("nodesRecoverable").isBoolean().booleanValue();
 
-            JSONObject infrastructurePluginDescriptorJson = jsonObject.get("infrastructurePluginDescriptor").isObject();
-            String infrastructurePluginName = infrastructurePluginDescriptorJson.get("pluginName")
-                                                                                .isString()
-                                                                                .stringValue();
-            PluginDescriptor infrastructurePluginDescriptor = getPluginDescriptor(infrastructurePluginDescriptorJson,
-                                                                                  infrastructurePluginName);
-
-            JSONObject policyPluginDescriptorJson = jsonObject.get("policyPluginDescriptor").isObject();
-            String policyPluginName = policyPluginDescriptorJson.get("pluginName").isString().stringValue();
-            PluginDescriptor policyPluginDescriptor = getPluginDescriptor(policyPluginDescriptorJson, policyPluginName);
+            PluginDescriptor infrastructurePluginDescriptor = getInfrastructurePluginDescriptor(jsonObject);
+            PluginDescriptor policyPluginDescriptor = getPolicyPluginDescriptor(jsonObject);
 
             return new NodeSourceConfiguration(nodeSourceName,
                                                nodesRecoverable,
                                                infrastructurePluginDescriptor,
                                                policyPluginDescriptor);
         } catch (RuntimeException e) {
-            throw new ImportException("The imported node source has incorrect parameters.", e);
+            throw new IllegalArgumentException("The imported node source has incorrect parameters.", e);
         }
     }
 
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)